quill.js

```diff
@@ -1,109 +1,28 @@
 import Quill from './core';
 
-import { AlignClass, AlignStyle } from './formats/align';
-import {
-  DirectionAttribute,
-  DirectionClass,
-  DirectionStyle,
-} from './formats/direction';
-import Indent from './formats/indent';
-
-import Blockquote from './formats/blockquote';
-import Header from './formats/header';
-import List from './formats/list';
-
-import { BackgroundClass, BackgroundStyle } from './formats/background';
-import { ColorClass, ColorStyle } from './formats/color';
-import { FontClass, FontStyle } from './formats/font';
-import { SizeClass, SizeStyle } from './formats/size';
-
 import Bold from './formats/bold';
 import Italic from './formats/italic';
-import Link from './formats/link';
-import Script from './formats/script';
-import Strike from './formats/strike';
 import Underline from './formats/underline';
 
-import Formula from './formats/formula';
-import Image from './formats/image';
-import Video from './formats/video';
-
-import CodeBlock, { Code as InlineCode } from './formats/code';
-
-import Syntax from './modules/syntax';
-import Table from './modules/table';
-import Toolbar from './modules/toolbar';
-
-import Icons from './ui/icons';
-import Picker from './ui/picker';
-import ColorPicker from './ui/color-picker';
-import IconPicker from './ui/icon-picker';
-import Tooltip from './ui/tooltip';
-
-import BubbleTheme from './themes/bubble';
-import SnowTheme from './themes/snow';
-
-Quill.register(
-  {
-    'attributors/attribute/direction': DirectionAttribute,
-
-    'attributors/class/align': AlignClass,
-    'attributors/class/background': BackgroundClass,
-    'attributors/class/color': ColorClass,
-    'attributors/class/direction': DirectionClass,
-    'attributors/class/font': FontClass,
-    'attributors/class/size': SizeClass,
-
-    'attributors/style/align': AlignStyle,
-    'attributors/style/background': BackgroundStyle,
-    'attributors/style/color': ColorStyle,
-    'attributors/style/direction': DirectionStyle,
-    'attributors/style/font': FontStyle,
-    'attributors/style/size': SizeStyle,
-  },
-  true,
-);
+/**
+ * This file defines which formatting modules should be included
+ * into the final build of Quill.
+ * Currently only Bold, Italic and Underline are allowed
+ * Disabling modules in the `formats` setting of Quill options
+ * (client side) does not fully blocks their usage.
+ * For example, having background color module registered here will allow
+ * pasted content with the background color even if `formats` array
+ * does not have BG color formatting enabled.
+ *
+ * For the full list of modules originally included with Quill
+ * please see quill.reg.original.js
```

Comment:
I wonder if this could be done in client code: importing `quill/core` configuring it only with modules that are used in the specific client.

I guess there's always a risk of one place importing top-level `quill` and having `Quill` be configured with additional modules that are unwanted. Client project could block-list that module I believe.

ref:

https://www.staging-typescript.org/tsconfig#files
https://www.staging-typescript.org/tsconfig#exclude